extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,261 @@
+      dev().error(
+        TAG,
+        'The first child must be a heading element <h1>, <h2>, or <h3>'
+      );
+    }
+
+    const prompt = document.createElement(promptInput.tagName);
+    prompt.textContent = promptInput.textContent;
+    prompt.classList.add('i-amp-story-quiz-prompt');
+    promptInput.remove();
+
+    const options = toArray(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      dev().error(TAG, 'Improper number of options');
+    }
+
+    this.quizEl_
+      .querySelector('.i-amp-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      dev().error(TAG, 'Too many children');
+    }
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the quiz element.
+   *
+   * @param {HTMLOptionElement} option
+   * @param {number} index
+   * @private
+   */
+  configureOption_(option, index) {
+    const convertedOption = buildOptionTemplate(option);
+
+    // Fill in the answer choice
+    convertedOption.querySelector(
+      '.i-amp-story-quiz-answer-choice'
+    ).textContent = answerChoiceOptions[index];
+
+    // Transfer the option information into a span then remove the option
+    const optionText = document.createElement('span');
+    optionText.setAttribute('class', 'i-amp-story-quiz-option-text');
+    optionText.textContent = option.textContent;
+    convertedOption.append(optionText);
+
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    convertedOption.setAttribute(
+      'id',
+      `${this.element.id}+${answerChoiceOptions[index]}`
+    );
+
+    convertedOption.setAttribute(
+      'on',
+      `tap:${this.element.id}.triggerInteraction`
+    );
+
+    // Add the option to the quiz element
+    this.quizEl_
+      .querySelector('.i-amp-story-quiz-option-container')
+      .appendChild(convertedOption);
+  }
+
+  /**
+   * Attaches functions to each option to handle state transition.
+   *
+   * @private
+   */
+  initializeListeners_() {
+    // Configure the response action
+    const actions = [
+      {tagOrTarget: 'AMP-STORY-QUIZ', method: 'triggerInteraction'},
+    ];
+    this.storeService_.dispatch(Action.ADD_TO_ACTIONS_WHITELIST, actions);
+
+    this.registerAction('triggerInteraction', actionInvocation => {
+      this.handleOptionSelection_(actionInvocation.caller);
+    });
+
+    // Add a listener for changes in the RTL state
+    this.storeService_.subscribe(
+      StateProperty.RTL_STATE,
+      rtlState => {
+        this.onRtlStateUpdate_(rtlState);
+      },
+      true /** callToInitialize */
+    );
+
+    // Add a click listener to the element to trigger the action via tapping the prompt
+    this.quizEl_.addEventListener('click', e => {
```

Comment:
Nit: please extract the listener to its own handler, like the `handleOptionSelection_` :)